lib/unison-prelude/src/Unison/Prelude.hs

```diff
@@ -233,6 +237,31 @@ writeUtf8 fileName txt = do
     Handle.hSetEncoding handle IO.utf8
     Text.hPutStr handle txt
 
+-- | Atomically prepend some text to a file
+prependUtf8 :: FilePath -> Text -> IO ()
+prependUtf8 fileName txt = do
+  let withTempFile tmpFilePath tmpHandle = do
```

Comment:
I just copied this from OutputMessages, but am wondering if we can't use `UnliftIO.withSystemTempFile` instead of writing our own masking?

Reply:
I tried to find the original author but the Github blame facilities aren't great.

Reply:
@tstat I couldn't figure out the original author to ask, but was it you / do you know?

Reply:
Chris and I caught up on slack yesterday, but copying the answer here:

You could replace https://github.com/unisonweb/unison/blob/trunk/unison-cli/src/Unison/CommandLine/OutputMessages.hs#L2622-L2628 with 

```
UnliftIO.withTempFile (FilePath.takeDirectory path) "unison-scratch" withTempFile
```
That code is very similar. 

We don't want to use `withSystemTempFile` and `renameFile` though, because the system temp dir is often mounted with a different file system (like tmpfs) and we can't do an atomic rename operation across devices.